File: src/exercises/most-dsa/dota2-senate/cpp/solution.cpp
class Solution {
public:
    string predictPartyVictory(string senate) {
        int cnt = 0, i = 0;

        while (i < senate.size()) {
            char c = senate[i];
            if (c == 'R') {
                if (cnt < 0) {
                    senate.push_back('D');
                }
                cnt++;
            } else {
                if (cnt > 0) {
                    senate.push_back('R');
                }
                cnt--;
            }
            i++;
        }

        return cnt > 0 ? "Radiant" : "Dire";
    }
};

File: src/exercises/most-dsa/dota2-senate/rust/solution.rs
impl Solution {
    pub fn predict_party_victory(senate: String) -> String {
        let mut senate: Vec<char> = senate.chars().collect();
        let mut cnt: i32 = 0;
        let mut i = 0;

        while i < senate.len() {
            let c = senate[i];
            if c == 'R' {
                if cnt < 0 {
                    senate.push('D');
                }
                cnt += 1;
            } else {
                if cnt > 0 {
                    senate.push('R');
                }
                cnt -= 1;
            }
            i += 1;
        }

        if cnt > 0 {
            "Radiant".to_string()
        } else {
            "Dire".to_string()
        }
    }
}
